File: smart_reminder/frontend/lib/models/reminder_model.dart
enum RepeatType { none, daily, weekly, monthly }

class ReminderModel {
  final String id;
  String title;
  String? description;
  DateTime dateTime;
  bool isCompleted;
  RepeatType repeatType;
  String? noteId;

  ReminderModel({
    required this.id,
    required this.title,
    this.description,
    required this.dateTime,
    this.isCompleted = false,
    this.repeatType = RepeatType.none,
    this.noteId,
  });

  ReminderModel copyWith({
    String? id,
    String? title,
    String? description,
    DateTime? dateTime,
    bool? isCompleted,
    RepeatType? repeatType,
    String? noteId,
  }) {
    return ReminderModel(
      id: id ?? this.id,
      title: title ?? this.title,
      description: description ?? this.description,
      dateTime: dateTime ?? this.dateTime,
      isCompleted: isCompleted ?? this.isCompleted,
      repeatType: repeatType ?? this.repeatType,
      noteId: noteId ?? this.noteId,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'dateTime': dateTime.toIso8601String(),
      'isCompleted': isCompleted,
      'repeatType': repeatType.index,
      'noteId': noteId,
    };
  }

  factory ReminderModel.fromMap(Map<String, dynamic> map) {
    return ReminderModel(
      id: map['id'],
      title: map['title'],
      description: map['description'],
      dateTime: DateTime.parse(map['dateTime']),
      isCompleted: map['isCompleted'] ?? false,
      repeatType: RepeatType.values[map['repeatType'] ?? 0],
      noteId: map['noteId'],
    );
  }
}

File: smart_reminder/backend/app/models/reminder_model.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Text, Boolean, DateTime, ForeignKey, Enum
from app.database.base import Base
import enum


class RepeatType(str, enum.Enum):
    none    = "none"
    daily   = "daily"
    weekly  = "weekly"
    monthly = "monthly"


class Reminder(Base):
    __tablename__ = "reminders"

    id          = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id     = Column(String, ForeignKey("users.id"), nullable=False)
    title       = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    date_time   = Column(DateTime, nullable=False)
    is_completed = Column(Boolean, default=False)
    repeat_type = Column(Enum(RepeatType), default=RepeatType.none)
    note_id     = Column(String, nullable=True)
    created_at  = Column(DateTime, default=datetime.utcnow)